src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp

```
@@ -88,28 +96,25 @@ struct EvolutionMetavars : public GeneralizedHarmonicTemplateBase<3, UseLts> {
       "formulation,\n"
       "on a domain with a single horizon and corresponding excised region"};
 
-  struct ApparentHorizon
-      : tt::ConformsTo<intrp::protocols::InterpolationTargetTag> {
-    using temporal_id = ::Tags::Time;
-    using tags_to_observe = ::ah::tags_for_observing<Frame::Inertial>;
-    using surface_tags_to_observe = ::ah::surface_tags_for_observing;
-    using compute_vars_to_interpolate = ah::ComputeHorizonVolumeQuantities;
-    using vars_to_interpolate_to_target =
-        ::ah::vars_to_interpolate_to_target<volume_dim, ::Frame::Inertial>;
-    using compute_items_on_target =
-        ::ah::compute_items_on_target<volume_dim, Frame::Inertial>;
-    using compute_target_points =
-        ah::TargetPoints::ApparentHorizon<ApparentHorizon, ::Frame::Inertial>;
-    using post_interpolation_callbacks =
-        tmpl::list<intrp::callbacks::FindApparentHorizon<ApparentHorizon,
-                                                         ::Frame::Inertial>>;
+  struct ApparentHorizon : tt::ConformsTo<ah::protocols::HorizonMetavars> {
+    using time_tag = ah::Tags::ObservationTime<0>;
+
+    using frame = ::Frame::Inertial;
+
+    using horizon_find_callbacks = tmpl::list<
+        ah::callbacks::ObserveTimeSeriesOnHorizon<
+            ::ah::tags_for_observing<Frame::Inertial>, ApparentHorizon>,
+        ah::callbacks::ObserveFieldsOnHorizon<::ah::surface_tags_for_observing,
+                                              ApparentHorizon>>;
     using horizon_find_failure_callbacks =
-        tmpl::list<intrp::callbacks::IgnoreFailedApparentHorizon>;
-    using post_horizon_find_callbacks = tmpl::list<
-        intrp::callbacks::ObserveTimeSeriesOnSurface<tags_to_observe,
-                                                     ApparentHorizon>,
-        intrp::callbacks::ObserveSurfaceData<
-            surface_tags_to_observe, ApparentHorizon, ::Frame::Inertial>>;
+        tmpl::list<ah::callbacks::FailedHorizonFind<ApparentHorizon, true>>;
```

Again, do we want to be ignoring errors?  Same for ST.